### DISCUSSION TITLE
kein Erfolg beim flashen

### DISCUSSION THREAD

**Discussion by boctus** [tbnobody/OpenDTU#1342] [Q&A]
Hallo Leute,
seit Tagen versuche ich ein ESP32 zu flashen. Az-delivery ESP31 Wroom 32, USB-Kabel, Windows, flash-download-tool 3.9.5.,
alle Einstellungen wie beschriebn gemacht, die openDTU.generic.factory.bin runtergeladen, dann ERASE und den boot-button auf dem ESP gedrückt. Funtzt neet. Button gedückt halten während man ERASE drückt, länger drücken, etc. Im Gerätemanger wird meine COM9 mit dem Silicon Laps Bridge angezeigt, Den Treiber hatte ich vorher installieren müssen.
Es kommt immer dieselbe Fehlermeldung in der CommandoFenster. Ich verstehe es einfach nicht.
Kann mir jemand sagen was ich falsch mache?
Danke.
Martin

![fl](https://github.com/tbnobody/OpenDTU/assets/126296759/1d05e77c-6f46-41a4-821b-3cbcf14bcc87)

**Comment by Tiese** [comment]
Hallo boctus,
was mir einfallen würde, versuch mal die Flash-Software mit Administrator-Rechte zu öffnen.
Vielleicht klappt es dann?!

**Comment by DPO99** [comment]
EDIT²**

Ich habs mal nachgestellt. Die 3.9.5 verhält sich m.E. etwas anders aber das Vorgehen ist eigentlich identisch:

-Flasher Tool starten und Auswahl "Esp32"
-ESP32 anstecken, COM-Port, Einstellungen etc. prüfen Firmware reinpacken usw. dann klick auf "Erase" und warten!
-Sobald das Flasher Tool auf "Download" steht, drückst du den "Boot" Button auf dem ESP und warten!
-Sobald das Flasher Tool auf "Finish" steht, drückst du auf "Start" und warten! (jetzt dauerts etwas aber es gibt einen Ladebalken)
-Sobald das Flasher Tool erneut auf "Finish" steht, einmal Strom weg und wieder ran, dann sollte der AP offen sein.
-Flasher Tool kann geschlossen werden

@tbnobody ggf. macht es Sinn, die Anleitung etwas nachzuschärfen?

**Comment by boctus** [comment]
Danke an alle. Es klappt nicht. Die Adminrechte hatte ich schon auf dem PC .
Habe auch probiert, den "Boot" button gedrückt zu halten während ich das USB Kabel einstecke ,dann auf Erase und dann Button los lassen. geht aber auch nicht. Es kommt immer wieder die Meldung: "no log file output".
Ich habe es jetzt über das Mac iOS Terminal gelöst und das esptool.py benutzt. Hatte die Anleitung auf YT gefunden.
Trotzdem allen Danke. Weiß bis heute nicht woran es liegt. Ist wieder so ein Windows-Ding. Vielleicht wurde der Treiber von der Bridge nicht richtig installiert...?

**Comment by jblond4711** [comment]
Hi,

möchte mich her mal mit dranhängen. Nachdem heute nach dem OTA Update nichts mehr ging war ich gezwungen neu zu flashen.
Dann wollte ich wie gewohnt mit dem flash-download-tool  3.9.3 loslegen habe aber dann auf der ersten Seite gesehen das es eine Neue Vorgehensweise gibt.
Leider bin ich sowohl mit der 3.9.3 als auch mit der 3.9.5 kläglich gescheitert.
Ich möchte aber noch erwähnen das es mit der „alten“ Methode immer hervorragend geklappt hat.
Mit der Webversion vom esp_flasher hat es dann zum Glück letztendlich dann doch noch geklappt.
Mich würde aber schon interessieren weshalb es mit dem flash-download-tool nicht mehr funktioniert.

Edit: Die v23.12.28 generic_esp32_factory.bin ließ sich komischerweise mit dem flash-download-tool  3.9.3 ohne Probleme flashen.

**Comment by Sorbat** [comment]
Ich hatte gestern auch so meine Probleme beim flashen, aber irgendwann ging es mit dem Tool.

danach war der Zugriff über den pc aber extrem träge, ein seitenaufbau dauerte bis zu 10 Sekunden. (Ist das normal?)
Über das Handy kam ich gar nicht drauf

**Comment by tuemler24** [comment]
Bei mir war das  gleiche Problem. Ich konnte mit dem ESP32 Flash Tool keinen neuen ESP32 Wroom 32 flashen. Über den Geräte Manger (Windows) habe ich gesehen, dass der CP2102 USB to UART Bridge Virtual COM Port Driver nicht geladen wurde. Ich habe den Treiber von folgender WEB Seite geholt: https://www.silabs.com/developers/usb-to-uart-bridge-vcp-drivers?tab=downloads
Diesen Treiber CP210x Universal Windows Driver habe ich in ein Verzeichis extrahiert und dann über den Geräte Manger (CP2102 Treiber) neu installiert. Danach habe ich den Com Port ermittelt und im ESP 32 Flash Tool eingetragen. Danach konnte ich mit Windows den ESP32 ohne Probleme Flashen.
PS: ich hoffe diese etwas ausführliche Erklärung kann hilfreich sein.

**Reply by Xtravaganz** [reply]
Genau daran liegt es, danach geht es ohne Probleme!

**Comment by diane-alexandra** [comment]
Hallo Leute, wie lange dauert es flashen? Mein PC rödelt jetzt schon über 10 Std. und schreibt nur "Punkte" Zeilen über Zeilen.
Gruß Diane

**Reply by DPO99** [reply]
Also das reine flashen der kompilierten Firmware dauert normalerweise 2 Minuten.

**Comment by diane-alexandra** [comment]
Im Geräte Manager steht RTL8191S WLAN Adapter aber ohne Zugriff. Was muss ich tun um den am laufen zu bekommen?

**Comment by Solarteur** [comment]
Hast Du 

https://www.opendtu.solar/firmware/ und die dort verlinkten Hinweise

beachtet?

Welche Hardware willst Du denn flashen?

RTL8191S WLAN Adapter klingt "komisch".

**Reply by diane-alexandra** [reply]
![Manager](https://github.com/user-attachments/assets/fd45edb0-16b4-432b-bad0-8645ddcb27e8)
Habe kein Zugriff auf die OpenDTU

**Comment by Solarteur** [comment]
Bei mir sieht ein ESP32 im Gerätemanager beispielsweise so aus - an COM3:
![grafik](https://github.com/user-attachments/assets/88392f88-1d39-4093-8469-6075e5924f87)

Bei einem anderen USB-to-UART Chip kann da auch etwas anderes stehen.

Was wird denn bei Dir unter "Anschlüsse (COM & LPT)" angezeigt?

Der "RTL8191S WLAN Adapter" könnte ein Treiber-Problem haben.

**Reply by diane-alexandra** [reply]
![Manager](https://github.com/user-attachments/assets/a6e3ccc3-86cc-4be8-bac5-faeed5ce5921)
Nur die beiden, mehr steht da nicht drin und ESP-32 via Com1 flashen geht nicht

**Comment by Solarteur** [comment]
Es sieht so aus, als hätte das ESP32-Board keine Verbindung zum PC. Schon ein anderes USB-Kabel ausprobiert? 
Nochmals die Frage zur OpenDTU-Hardware: Selbstbau (welche Komponenten?) oder "fertig gekauft".

**Reply by diane-alexandra** [reply]
Ja, auch mit 2 Leitungen keine Reaktion. Ein fertig erstelltes Teil von Nordic ( aber privat gekauft )

**Reply by Solarteur** [reply]
Da fällt mir leider auch nichts mehr ein. Vielleicht hat der Verkäufer eine Idee.

**Reply by diane-alexandra** [reply]
Hi Solarteur, ein Freund war gerade hier, hat sich an mein PC gesetzt und während ich Kaffee machte, Ratze-Fatze, openDTU online. Dank dir für deine Bemühungen mir zu helfen. Gruß Diane

**Comment by wegomyway** [comment]
Musste meinen OpenDTU AllianzApps flashen, weil der laufende Meter es vollbracht hat meinen Zettel mit dem neuen Passwort in den Grill zu feuern.
Kann mich den Usern hier nur anschließen, unter Windows, trotz der Treiber und den guten Anleitungen habe ich immer „no log File output“, egal was ich einstelle, welches *.bin ich nehme. Hat nicht geklappt. Die Online-Variante hat dann auf Anhieb funktioniert und nun ist erstmal wieder gut, weil testweise ich mich mal probiert habe über das Web einzuloggen mit den Standard-Werten. Hat geklappt. Nochmals geflasht und morgen geht es wieder zum Bestimmungsort und wird erneut eingerichtet. Mit zwei Hoymiles, den 600er und 400er. Danach dann EverHome ECO-Tracker an den 2RZ ran und die Hoymiles beide da einbinden.

**Comment by wegomyway** [comment]
Musste meinen OpenDTU AllianzApps flashen, weil der laufende Meter es vollbracht hat meinen Zettel mit dem neuen Passwort in den Grill zu feuern.
Kann mich den Usern hier nur anschließen, unter Windows, trotz der Treiber und den guten Anleitungen habe ich immer „no log File output“, egal was ich einstelle, welches *.bin ich nehme. Hat nicht geklappt. Die Online-Variante hat dann auf Anhieb funktioniert und nun ist erstmal wieder gut, weil testweise ich mich mal probiert habe über das Web einzuloggen mit den Standard-Werten. Hat geklappt. Nochmals geflasht und morgen geht es wieder zum Bestimmungsort und wird erneut eingerichtet. Mit zwei Hoymiles, den 600er und 400er. Danach dann EverHome ECO-Tracker an den 2RZ ran und die Hoymiles beide da einbinden.

**Comment by OttiNC1** [comment]
Immer wieder Probleme beim flashen. Jetzt 8 Kabel durchprobiert. Mit Einem hat es dann geklappt. Immer gleiche Vorgehensweise! Kabel jetzt markiert.
Datei: opendtu-onbattery-generic_esp32_4mb_no_ota.factory.bin  (4MB ESP)
Zu früh gefreut! Mit einem anderen ESP 32 WROOM ging es nicht mehr...
Dann nochmal probiert. Tool V 3.9.3 dann lief es auch mit dem ESP32, wo es nicht klappte.
Nach der Anleitung DPO99 (weiter zurück). Manchmal mußte ich auch mehrmals "START" drücken, teilweise die Erkennung des COM durch Windows abwerten (Bling-Ton). Der Fortschritt- Balken muß kommen.
Das ist ein Krampf!
